test/check-machines-lifecycle

```diff
@@ -49,6 +49,10 @@ class TestMachinesLifecycle(VirtualMachinesCase):
 
         return user_name
 
+    def testBasicNonrootUserUnprivileged(self):
+        user = self.createUser(user_group=None)
+        self._testBasic(user, False, True)
```

Comment:
I don't quite understand how this can behave exactly the same way as testBasicWheelUserUnprivileged(). The latter ought to get superuser privileges, this one shouldn't. There must be some measurable difference. Actually, *this* new test case makes sense, I don't understand the existing testBasicWheelUserUnprivileged(). @yunmingyang , @skobyda , any thoughts here?

Reply:
So in order for wheel group user to have special privileges, you have to amend libvirt polkit policies. Without them, which is the case in our tests, wheel group user is basically any user without access to libvirt daemon. So I think testBasicNonrootUserUnprivileged is doing exactly the same as testBasicWheelUserUnprivileged.

What I think is missing in our tests instead is the extension for testBasicWheelUserUnprivileged which grants wheel group user priviliged access to qemu:///system by ammending polkit policies and see if it works properly. However there are currently som bugs with polkit, which need to be fixed first (e.g. https://bugzilla.redhat.com/show_bug.cgi?id=1820493), so that would be outside of the scope of this PR.

Reply:
So in summary, I think testBasicNonrootUserUnprivileged doesn't add anything new compared to testBasicWheelUserUnprivileged. But I'm up for deleting testBasicWheelUserUnprivileged, since its name is kinda misleading, and reintroducing it back, with extended tests for privileged access for wheel group once the polkit issues are fixed.

Reply:
What I meant is that a `wheel` user can run `sudo` and thus should be able to connect to libvirt as root. A test *can* disable this with `login_and_go(..., superuser=False)` option, but this test doesn't. That's where my confusion comes from -- why does this test not simply connect as root?

Reply:
So I went back in git history and found that this seems to be rewritten from older test:

```
-    def testWheelUserOperationWithVm(self):
-        name = 'staticvm'
-        vm_args = self.create_vm(name, wait=True)
-
-        # A admin user in the wheel group is allowed to see the VM and
-        # can do operations on it, but only with escalated privileges.
-
-        self.non_root_user_operations(user_group='wheel', privilege=False,
-                                      operations=False)
-
-        self.non_root_user_operations(user_group='wheel', privilege=True,
-                                      operations=True, vm_args=vm_args)

```

Reply:
So yeah I was wrong, you don't need polkit setup for that. But if you login with user with wheel privileges, you right now won't see any VMs (reproducible with `[wheeluser@fedora cockpit]$virsh -c qemu:///system list --all`) but once you authorize, you will get access to qemu:///system (reporducible with `[wheeluser@fedora cockpit]$sudo virsh -c qemu:///system list --all`). So I think the tests should reflect this behaviour

Reply:
I found we use self.eval_js('window.localStorage.setItem("superuser:%s", "%s");' % (user, "any" if superuser else "none")) to implement "superuser" in login_and_go. But it is "sudo" by window.localStorage.getItem("superuser:test_wheel") if changing the privilege in UI. Should modify "any" to "sudo"?

Reply:
Ah I'm not really familiar with that code, or where and how it gets read by cockpit during authorization process. @martinpitt any idea?

Reply:
I'm afraid I don't understand @yunmingyang 's question. The `superuser` flag in `login_and_go()` determines whether the test will do the equivalent of clicking the "Limited access" button in the UI, so that the cockpit session will start a root bridge (through sudo) and hence all `{ superuser: "try" }` or `{ superuser: "require" }` cockpit API calls will run as root. Being in the `wheel` group allows a user to become root through sudo.

In other words, if you log in as `test_wheel_user` with `superuser=False`, this will behave like a normal unprivileged user. With `superuser=True` you will be able to do any administrative action including libvirt (this is totally independent of polkit, it just connects to libvirtd as root).

Reply:
I mean, in the test, it seems to use "window.localStorage.setItem("superuser:%s", "%s")" in the browser console, before click "Log in", to do the equivalent of clicking the "Limited access" button in the UI. And if we use a wheel group user like test_wheel_user, to get "Administrative access", in test, it seems to use "window.localStorage.setItem("superuser:test_wheel_user", "any")". But if we get "Administrative access" by clicking ""Limited access", the value of "superuser:test_wheel_user" getting by window.localStorage.getItem("superuser:test_wheel_user") is "sudo", not "any". So my question is, is there some different when setting "superuser:test_wheel_user" to "sudo" or "any"?

Reply:
@yunmingyang : not in a meaningful way in this context, no. There used to be a fallback for running `pkexec cockpit-bridge` instead of `sudo cockpit-bridge`, but in practice, `any` and `sudo` are equivalent for "connect with browser" mode.